go/vt/vtgate/vindexes/lookup.go

```diff
@@ -286,3 +350,15 @@ func (lu *LookupUnique) Delete(vcursor VCursor, rowsColValues [][]sqltypes.Value
 func (lu *LookupUnique) MarshalJSON() ([]byte, error) {
 	return json.Marshal(lu.lkp)
 }
+
+func numericUint64(input sqltypes.Value) ([]byte, error) {
+	// If attempting to cherry-pick this into a build <= 6.0 this needs to be
+	// rewritten as sqltypes.ToUint64
+	v, err := evalengine.ToUint64(input)
+	if err != nil {
+		return nil, fmt.Errorf("numericUint64: couldn't parse bytes: %v", err)
+	}
+	vBytes := make([]byte, 8)
+	binary.LittleEndian.PutUint64(vBytes, v)
```

Comment:
I want to call out that I'm forcing LittleEndian here bc of the key Destination ID type declaration (`type DestinationKeyspaceID []byte`) and my understanding of how keyspace IDs work. If that's not correct and I should use host-order or force BigEndian lmk

Reply:
Ah, it's in the obvious place:

```
// KeyRangeContains returns true if the provided id is in the keyrange.
func KeyRangeContains(kr *topodatapb.KeyRange, id []byte) bool {
	if kr == nil {
		return true
	}
	return bytes.Compare(kr.Start, id) <= 0 &&
		(len(kr.End) == 0 || bytes.Compare(id, kr.End) < 0)
}
```

which actually leaves me thinking that it should be BigEndian so I'd def like a consult here.

Reply:
And from [the keyrange docs](https://vitess.io/docs/reference/features/sharding/#key-ranges-and-partitions) it implies that the ranges are MSB:

> Vitess always converts sharding keys to a left-justified binary string for computing a shard. This left-justification makes the right-most zeroes insignificant and optional. Therefore, the value 0x80 is always the middle value for sharding keys. So, in a keyspace with two shards, sharding keys that have a binary value lower than 0x80 are assigned to one shard. Keys with a binary value equal to or higher than 0x80 are assigned to the other shard.

Given this I'm going to update the impl to use `BigEndian` and leave the thread open in case this is a misread of the docs.